api/gateway_test.go

```diff
@@ -51,18 +50,22 @@ func TestPeering(t *testing.T) {
 		t.Fatal("/gateway/peer/add did not add peer", peer2.netAddress())
 	}
 
-	// Create a third peer that bootstraps to the first peer and check that it
+	// Create a third peer that bootstraps to the first peer, and check that it
 	// reports the others as peers.
 	peer3 := peer1.addPeer("TestPeering3")
 	peer3.getAPI("/gateway/status", &info)
 	if len(info.Peers) != 2 {
-		t.Fatal("bootstrap peer did not share its peers")
+		t.Fatal("bootstrap peer did not share its peers", info)
 	}
 
-	// peer2 should have received peer3 via peer1. Note that it does not have
-	// peer1 though, because /gateway/add does not contact the added peer.
+	// When peer3 bootstraps to peer1, it should learn about peer2 and
+	// announce to it.
 	peer2.getAPI("/gateway/status", &info)
-	if len(info.Peers) != 1 || info.Peers[0] != peer3.netAddress() {
+	for len(info.Peers) != 1 {
+		time.Sleep(time.Millisecond)
+		peer2.getAPI("/gateway/status", &info)
+	}
```

Comment:
We should probably add a gateway subscription channel here instead of busywaiting.